actionview/lib/action_view/helpers/tag_helper.rb

```diff
@@ -344,6 +344,10 @@ def tag(name = nil, options = nil, open = false, escape = true)
       #   <% end -%>
       #    # => <div class="strong">Hello world!</div>
       def content_tag(name, content_or_options_with_block = nil, options = nil, escape = true, &block)
+        html_tag_regex = /^[a-zA-Z][\w-]*$/
```

Comment:
This regex allows,
- Letters (A-Z and a-z): Tag names can consist of uppercase and lowercase letters.
- Digits (0-9): Tag names can include numbers, but they cannot start with a digit.
- Hyphens ("-")
- Underscores ("_")

Reply:
@flavorjones Let me know your thoughts on this.